### ISSUE TITLE
Docker Container for Support between tf-io and tf-serving.

### ISSUE THREAD

**Issue by Ouwen** [tensorflow/serving#1411]
Using the tf.io ops in the tf.serving ecosystem would be a large development convenience and likely decrease inference latency. Can there be an official docker build or documentation to integrate tensorflow-io with tensorflow-serving?

Related issue on [tensorflow/io issue #414](https://github.com/tensorflow/io/issues/414)

**Comment by unclepeddy** [comment]
It's a reasonable request. For the short term, you can build ops into modelserver using this [guide](https://github.com/tensorflow/serving/blob/master/tensorflow_serving/g3doc/custom_op.md). 
However, as you noted, the dev experience would be much neater if you could just download a docker image with the op already built in. We're looking into how we want to support custom ops internally.
2 questions:
Any particular ops in tf.io? or all?
And why do you expect decreased inference latency depending on whether you link it into the model server vs. we release a package with it linked in?

**Comment by Ouwen** [comment]
@unclepeddy there wouldn't be a decreased inference latency via linking with the model server. Prior to model server custom ops, we would parse files into protobuf and grpc request tf-serving (quite inefficient). This would be mainly a dev experience improvement.

Ideally, all tf.io ops could be supported. My main interest is in the `dicom` operations.

**Comment by unclepeddy** [comment]
I see - so in order to have officially support tf.io ops we need to get a better understanding of their policies (I've asked a question in the issue you opened in their repo). 

To parallelize the work though, I can try to see if we would run into any technical challenges if we were to do this (for ex. integrating tf.text ops into serving has proven to be fairly challenging because of issues we run into when linking all their dependencies statically). Could you provide me with a test model that uses tf.io ops and example requests?

**Comment by Ouwen** [comment]
@unclepeddy yes I can send you a small test model and thanks for the help. I'll send it to your email `pedrampejman2010@gmail.com` unless there is another email you'd like for me to use.

**Comment by unclepeddy** [comment]
That works. Please ensure to include example requests as well. Thanks @Ouwen !

**Comment by Ouwen** [comment]
@marcelolerendegui

**Comment by tinder-michaelallman** [comment]
> It's a reasonable request. For the short term, you can build ops into modelserver using this guide.

Hi @unclepeddy. Thanks for the documentation referral. Unfortunately, despite the guidance I have not been able to successfully include tf-io in a custom model server build. I think the problems I've had are related to the particular way tf-io is built (as described in their README). If you have some time, could you look into how this can be accomplished and report back? I'm sure it would be much appreciated by all.

Cheers.

**Comment by lminer** [comment]
Is there any update to this? Has anyone managed to create a tensorflow-serving dockerfile with tfio support?

**Comment by singhniraj08** [comment]
@Ouwen,

Are you still looking for a resolution? We are planning on prioritising the issues based on the community interests. 
As a workaround you can follow [TF Serving custom Ops](https://www.tensorflow.org/tfx/serving/custom_op) guide to create custom TF Serving with tensorflow-io support. Thanks.

**Comment by gbildson** [comment]
Why would you "modularize" components but then not support their ops in tensorflow-serving?

**Comment by bharatjetti** [comment]
Hi @Ouwen  Thank you so much for showing interest! We truly appreciate your enthusiasm and support.
We regret to inform you that we are not accepting new implementation requests or feature additions at this time. Please feel free to fork the repository and submit pull requests if you have any suggestions for enhancements, problem fixes, or ideas. We’d be more than happy to review and collaborate! If you would like to contribute , please follow our [CONTRIBUTION GUIDELINES](https://github.com/tensorflow/serving/blob/master/CONTRIBUTING.md) here and submit a pull request.
Thank you again for your interest, and we look forward to seeing your contributions!

**Comment by jeelania10** [comment]
Closing this due to inactivity. Please feel free to reopen and post your comments (if you still have queries on this). Thank you!
